cmd/dockerd/daemon.go

```diff
@@ -53,6 +53,9 @@ type DaemonCli struct {
 	*daemon.Config
 	commonFlags *cli.CommonFlags
 	configFile  *string
+
+	api *apiserver.Server
+	d   *daemon.Daemon
```

Comment:
Instead of adding those two fields, I would rather add them as parameters to `setupConfigReloadTrap` (which in turn would pass them down to `reloadConfig`). Of course, that is unless they are being used elsewhere, but I don't see that being the case on first review.